main.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/crane"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+)
+
+func build(diskPath string) (v1.Image, error) {
+	// Add the disk.qcow2 file as a new layer
+	layer, err := tarball.LayerFromFile(diskPath, tarball.WithCompressedCaching)
+	if err != nil {
+		log.Fatalf("Error creating layer from file: %v", err)
+		return nil, err
+	}
+
+	// Append the layer to the base image
```

Comment:
@codingben not for this PR, just a general hint about code style: it is not helpful to explain what is happening in the line below if the line below is self-explanatory. https://bpoplauschi.github.io/2021/01/20/Clean-Code-Comments-by-Uncle-Bob-part-2.html is a great explanation.

Reply:
Alright thanks.